From: SAKAI Kiyotaka <ksakai@netwk.ntt-at.co.jp>
Newsgroups: fj.editor.emacs
Subject: Re: EDB
Date: 09 Dec 1998 11:32:52 +0900
Organization: NTT Advanced Technology Corp.
Lines: 54
Message-ID: <19981209113252U.ksakai@netwk.ntt-at.co.jp>
References: <sqlww4af8ar.fsf@etla3434.etl.go.jp>	<747nih$51f$1@sakyu.tottori-u.ac.jp>	<19981208161702Z.ksakai@netwk.ntt-at.co.jp>
NNTP-Posting-Host: kso.netwk.ntt-at.co.jp
Mime-Version: 1.0
Content-Type: Text/Plain; charset=iso-2022-jp
Content-Transfer-Encoding: 7bit
X-Trace: kso.netwk.ntt-at.co.jp 913170773 12896 192.168.9.2 (9 Dec 1998 02:32:53 GMT)
X-Complaints-To: news@kso.netwk.ntt-at.co.jp
NNTP-Posting-Date: 9 Dec 1998 02:32:53 GMT
In-Reply-To: <19981208161702Z.ksakai@netwk.ntt-at.co.jp>
X-Mailer: Mew version 1.94b2 on Emacs 20.3 / Mule 4.0 (HANANOEN)
X-Dispatcher: imput version 981124(IM104)

>> In article <19981208161702Z.ksakai@netwk.ntt-at.co.jp>, SAKAI Kiyotaka <ksakai@netwk.ntt-at.co.jp> writes:

> この EDB というのはどこから手に入るのでしょうか？

ある方から個人的に

  ftp://theory.lcs.mit.edu/pub/emacs/edb/

から手に入ると教えてもらいました。

それで、気になったので動作をさせてみたところ、

(defun db-copy-buffer-local-variables (buffer)
  "Copy the values of all of BUFFER's local variables into the current buffer."
  (let ((blv (in-buffer-simple buffer (buffer-local-variables)))
        pair symbol)
    (while (consp blv)
      (setq pair (car blv)
            symbol (car pair)
            blv (cdr blv))
      ;; nil and 0 can be bogus local variables; never copy buffer-undo-list
      (if (not (memq symbol '(0 nil buffer-undo-list)))
          (progn
            (if (not (symbolp symbol))
                (error "\"%s\" should be a symbol with value \"%s\""
                       symbol (cdr pair)))
            (make-local-variable symbol)
            (set symbol (cdr pair)))))))

というコードがあって、すべての buffer local 変数を make-local-variable 
を実行してからコピーするというコードで enable-multibyte-characters に
対しても make-local-variable を実行しているために引っかかっていること
が分かりました。

ですので、以下のように修正すればいいかと思います。

===================================================================
RCS file: RCS/db-util.el,v
retrieving revision 1.1
diff -u -r1.1 db-util.el
--- db-util.el	1998/12/09 02:26:58	1.1
+++ db-util.el	1998/12/09 02:27:39
@@ -557,7 +557,7 @@
 	    symbol (car pair)
 	    blv (cdr blv))
       ;; nil and 0 can be bogus local variables; never copy buffer-undo-list
-      (if (not (memq symbol '(0 nil buffer-undo-list)))
+      (if (not (memq symbol '(0 nil buffer-undo-list enable-multibyte-characters)))
 	  (progn
 	    (if (not (symbolp symbol))
 		(error "\"%s\" should be a symbol with value \"%s\""

-- 
酒井 清隆 (E-mail: ksakai@netwk.ntt-at.co.jp)


*** db-interfa.el.orig	Sun Jan 10 20:50:33 1999
--- db-interfa.el	Sun Jan 10 20:51:41 1999
***************
*** 434,439 ****
--- 434,440 ----
        (define-key database-edit-mode-map [mouse1] 'db-lucid-mouse-jump-to-point)
        (define-key database-edit-mode-map [mouse3] 'database-edit-mode-menu)
  
+       (load "db-summary")
        (define-key database-summary-mode-map [mouse1] 'db-lucid-mouse-jump-to-point)
  
        (defun db-lucid-mouse-jump-to-point (e)
